lighthouse-core/audits/large-javascript-dependencies.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * @license Copyright 2020 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+
+/**
+ * @fileoverview This audit checks a page for any unnecessarily large npm dependencies.
+ * These libraries can be replaced with functionally equivalent, smaller ones.
+ */
+
+'use strict';
+
+const BundlePhobiaStats = require('../lib/large-libraries.json');
+const Audit = require('./audit.js');
+const i18n = require('../lib/i18n/i18n.js');
+
+const UIStrings = {
+  /** Title of a Lighthouse audit that provides detail on large Javascript libraries that are used on the page that have better alternatives. This descriptive title is shown when to users when no known unnecessarily large libraries are detected on the page.*/
+  title: 'Avoids unnecessarily large JavaScript dependencies',
+  /** Title of a Lighthouse audit that provides detail on large Javascript libraries that are used on the page that have better alternatives. This descriptive title is shown when to users when some known unnecessarily large libraries are detected on the page.*/
+  failureTitle: 'Includes unnecessarily large JavaScript dependencies',
+  /** Description of a Lighthouse audit that tells the user why they should care about the large Javascript libraries that have better alternatives.. This is displayed after a user expands the section to see more. No character length limits. */
+  description: 'Large JavaScript libraries can lead to poor performance. ' +
+    'Prefer smaller, functionally equivalent libraries to reduce your bundle size.' +
+    '[Learn more](https://developers.google.com/web/fundamentals/performance/webpack/decrease-frontend-size#optimize_dependencies).',
+  /** Label for a column in a data table. Entries will be names of suggested libraries that are an alternative to what's currently being used. */
+  suggestion: 'Suggestion',
+  /** Label for a column in a data table. Entries will be the amount of bytes saved by switching libraries calculated based on the gzip values. */
+  savings: 'Potential Size Savings (Bytes)',
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+const Suggestions = {'moment': 'dayjs'};
```

Comment:
Update - the suggestions database now uses an array instead of a mapping. It's the approach that BundlePhobia takes e.g. [here](https://github.com/pastelsky/bundlephobia/blob/e0b041b6f79118d68f13f014d4d4bce03537de2f/server/middlewares/similar-packages/fixtures.js#L33). If Lighthouse detects a library that matches a library in one of the arrays, it'll compare that library's size against every other library in the array and only the smaller ones will be added as suggestions.

So idt we need a test specifically for this anymore but general tests in the works rn

Reply:
got it, yeah an exhaustive test wouldn't be needed anymore but definitely a specific test.

another useful test I think would be a jest snapshot of the audit result given the presence of every key in `library-suggestions.json`

Reply:
how do I add a snapshot test?

Reply:
loosely it's 

```js
const result = await LargeJsAudit.audit(...);
expect(result.details.items).toMatchInlineSnapshot();
```

Reply:
running just will then autopopulate a string inside `toMatchInlineSnapshot`